src/ShortestPaths/astar.jl

```diff
@@ -24,6 +24,12 @@ struct AStarResult{T, U<:Integer} <: ShortestPathResult
     dist::T
 end
 
+function ==(a::ShortestPaths.AStarResult, b::ShortestPaths.AStarResult)
+   return a.path == b.path && a.dist == b.dist
+end
+
+parents(::AStarResult) = throw(LightGraphs.NotImplementedError("parents"))
```

Comment:
Is that necessary? Do we actually have a parents function implementd for the supertype of `AStarResult`?

Reply:
yes. `AStarResult <: ShortestPathResult` and `parents(x::ShortestPathResult) = x.parents`.

Reply:
Ok then, might not be the most elegant thing to do though, maybe at one point we can solve this somehow with the type hierarchy and some traits.